src/smclarify/bias/metrics/common.py

```diff
@@ -175,6 +175,47 @@ def ensure_series_data_type(series: pd.Series, values: Optional[List[Any]] = Non
     raise ValueError("Data series is invalid or can't be classified as neither categorical nor continous.")
 
 
+def ensure_label_value_data_types_for_categorical(series: pd.Series, positive_label_values: List[Union[str, int, float]]) -> List:
+    """
+    Determines the type of the given data series and then do necessary type conversion to ensure the positive_lable_values
+    are of the same type as those in series.
+
+
+    Example problem when it helps:
+    The problem is that the `label_values_or_threshold` and the actual `label` values are not the same -
+    i.e. do not have the same type. This leads to customer facing errors when they pass numerical values
+    to `label_values_or_threshold` (for instance `[1, 2, 3]`) but having string values in the label column
+    of the dataset (for instance, `['1', '2', '3', '4', '5']`).
+
+    :param series: data for facet/label/predicted_label columns
+    :param positive_label_values: list of label values provided by user
+    :return: list of label values provided after the conversion (if any)
+    """
+    def _convert(items: List, _type: Callable):
+        try:
+            return [_type(item) for item in items]
+        except ValueError as e:
+            # int('1.0') raises a ValueError
+            if "invalid literal for int() with base 10" in str(e):
+                return [float(item) for item in items]
```

Comment:
can you handle the full case, otherwise this could return None? maybe make sure to raise an error with a descriptive error message.

Reply:
improved